Mathlib/Data/Real/EReal.lean

```diff
@@ -1291,6 +1463,30 @@ lemma left_distrib_of_nonneg {a b c : EReal} (ha : 0 ≤ a) (hb : 0 ≤ b) :
   nth_rewrite 1 [EReal.mul_comm]; nth_rewrite 2 [EReal.mul_comm]; nth_rewrite 3 [EReal.mul_comm]
   exact right_distrib_of_nonneg ha hb
 
+lemma left_distrib_of_nonneg_of_ne_top {x : EReal} (hx_nonneg : 0 ≤ x)
+    (hx_ne_top : x ≠ ⊤) (y z : EReal) :
+    x * (y + z) = x * y + x * z := by
+  by_cases hx0 : x = 0
+  · simp [hx0]
+  have hx_pos : 0 < x := hx_nonneg.lt_of_ne' hx0
```

Comment:
Since you won't need `hx0` any more, you could also do `replace hx0 : 0 < x`... to have one hypothesis less in the context.